### ISSUE TITLE
Running open telemetry agent as a sidecar

### ISSUE THREAD

**Issue by buddhike** [aws/copilot-cli#2496]
What do you think about the ability to run `otel/opentelemetry-collector-contrib` as a sidecar with its configuration in the manifest? Perhaps this could be a sidecar pattern that comes OOB.

In addition to that, if otel config has `awsxray` exporter, co-pilot should automatically adjust the permissions for underlying task role.

**Comment by huanjani** [comment]
@buddyspike, thanks for this feature request!

**Comment by PettitWesley** [comment]
OTel has an X-Ray exporter. Additionally, we have an AWS Distro of the collector: https://github.com/aws-observability/aws-otel-collector

https://gallery.ecr.aws/aws-observability/aws-otel-collector

**Comment by buddhike** [comment]
@PettitWesley, that is correct. The idea is to have a workflow in copilot to enable otel agent as a sidecar. This means automating a few things:

- Configuring otel agent container with desired config as a sidecar. Users would have to share their config via an ephemeral volume.
- Ensuring ECS task role has access to required internal services (xray for example)

Thoughts?
